### PR TITLE
Add The West Australian

### PR DESCRIPTION
**Pull request by MaxDall**
This PR is based on #614 and should be reviewed afterward.

### CODE REVIEW

File: src/fundus/publishers/au/__init__.py [hunk 1]
```diff
@@ -15,3 +16,13 @@ class AU(metaclass=PublisherGroup):
...
         ],
     )
+
+    WestAustralian = Publisher(
+        name="The West Australian",
+        domain="https://thewest.com.au/",
+        parser=WestAustralianParser,
+        sources=[
+            NewsMap("https://thewest.com.au/news-sitemap.xml"),
+            Sitemap("https://thewest.com.au/sitemap.xml", reverse=True),
```
Comment (addie9800) [inline src/fundus/publishers/au/__init__.py:26]: There's also a RSS feed containing the most recent articles: https://thewest.com.au/rss

File: src/fundus/publishers/au/west_australian.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        def publishing_date(self) -> Optional[datetime.datetime]:
+            return generic_date_parsing(self.precomputed.ld.bf_search("datePublished"))
+
+        @attribute
+        def authors(self) -> List[str]:
+            return generic_author_parsing(self.precomputed.ld.bf_search("author"))
+
+        @attribute
+        def topics(self) -> List[str]:
+            return generic_topic_parsing(self.precomputed.ld.bf_search("keywords"))
```
Comment (addie9800) [inline src/fundus/publishers/au/west_australian.py:67]: For some reason TheWestAustralian seems to include a lot of duplicate topics. I think we should filter them to be unique (maybe it makes sense to implement it in `generic_topic_parsing`?). This is already visible in the testcase or in [This](https://thewest.com.au/news/wa/tanya-plibersek-has-half-heartedly-supported-nigel-satterleys-contentious-north-stoneville-townsite-c-16087647) article. Also, they add the topic `premium` to their plus articles, I would consider filtering that as well

File: src/fundus/publishers/au/west_australian.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+)
+
+
+class WestAustralianParser(ParserProxy):
+    class V1(BaseParser):
+        _page_data_selector = XPath(
+            "string(//script[re:test(text(), 'window.PAGE_DATA')])",
+            namespaces={"re": "http://exslt.org/regular-expressions"},
+        )
+        _json_type_pattern = re.compile(r"^(?P<type>.*)=\s*{")
```
Comment (addie9800) [inline src/fundus/publishers/au/west_australian.py:24]: This seems to be leftover. At least I couldn't find any usage in the project files.

**Review by addie9800 — CHANGES_REQUESTED** [review]
Thanks for adding this publisher, seems like it was quite some effort. I only got some minor suggestions :)

File: src/fundus/publishers/au/west_australian.py (hunk 2 above)
Comment (MaxDall) [reply src/fundus/publishers/au/west_australian.py:66]: Good Point! I will add this in a separate PR

**Review by addie9800 — APPROVED** [review]

**Pull request merged**